File: app/src/main/java/com/vehiclebooking/data/dao/SearchRecordDao.java
package com.vehiclebooking.data.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;
import com.vehiclebooking.data.model.SearchRecordEntity;
import java.util.List;

@Dao
public interface SearchRecordDao {
    @Query("SELECT * FROM search_records")
    List<SearchRecordEntity> getAllSearchRecords();

    @Query("SELECT * FROM search_records WHERE phoneNumber = :phoneNumber AND timestamp = :timestamp LIMIT 1")
    SearchRecordEntity getSearchRecord(String phoneNumber, String timestamp);

    @Query("SELECT * FROM search_records WHERE status = :status")
    List<SearchRecordEntity> getSearchRecordsByStatus(String status);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertSearchRecord(SearchRecordEntity record);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertSearchRecords(List<SearchRecordEntity> records);

    @Update
    void updateSearchRecord(SearchRecordEntity record);

    @Delete
    void deleteSearchRecord(SearchRecordEntity record);

    @Query("DELETE FROM search_records WHERE phoneNumber = :phoneNumber AND timestamp = :timestamp")
    void deleteSearchRecord(String phoneNumber, String timestamp);

    @Query("DELETE FROM search_records")
    void deleteAllSearchRecords();
}

File: app/src/main/java/com/vehiclebooking/data/dao/SearchRecordDao.kt
package com.vehiclebooking.data.dao

import androidx.room.*
import com.vehiclebooking.data.model.SearchRecordEntity
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.first
import kotlinx.coroutines.runBlocking

@Dao
interface SearchRecordDao {
    @Query("SELECT * FROM search_records ORDER BY timestamp DESC")
    fun getAllSearchRecords(): Flow<List<SearchRecordEntity>>

    @Query("SELECT * FROM search_records WHERE phoneNumber = :phoneNumber AND timestamp = :timestamp LIMIT 1")
    suspend fun getSearchRecord(phoneNumber: String, timestamp: String): SearchRecordEntity?

    @Query("SELECT * FROM search_records WHERE status = :status ORDER BY timestamp DESC")
    fun getSearchRecordsByStatus(status: String): Flow<List<SearchRecordEntity>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertSearchRecord(record: SearchRecordEntity)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertSearchRecords(records: List<SearchRecordEntity>)

    @Update
    suspend fun updateSearchRecord(record: SearchRecordEntity)

    @Delete
    suspend fun deleteSearchRecord(record: SearchRecordEntity)

    @Query("DELETE FROM search_records WHERE phoneNumber = :phoneNumber AND timestamp = :timestamp")
    suspend fun deleteSearchRecord(phoneNumber: String, timestamp: String)

    @Query("DELETE FROM search_records")
    suspend fun deleteAllSearchRecords()

    @Query("SELECT COUNT(*) FROM search_records")
    fun getSearchRecordCount(): Flow<Int>
    
    // Blocking wrappers for Java compatibility
    fun getAllSearchRecordsBlocking(): List<SearchRecordEntity> = runBlocking {
        getAllSearchRecords().first()
    }
    
    fun getSearchRecordBlocking(phoneNumber: String, timestamp: String): SearchRecordEntity? = runBlocking {
        getSearchRecord(phoneNumber, timestamp)
    }
    
    fun getSearchRecordsByStatusBlocking(status: String): List<SearchRecordEntity> = runBlocking {
        getSearchRecordsByStatus(status).first()
    }
    
    fun insertSearchRecordBlocking(record: SearchRecordEntity) = runBlocking {
        insertSearchRecord(record)
    }
    
    fun updateSearchRecordBlocking(record: SearchRecordEntity) = runBlocking {
        updateSearchRecord(record)
    }
    
    fun deleteAllSearchRecordsBlocking() = runBlocking {
        deleteAllSearchRecords()
    }
}
